npm/vue/src/index.ts

```diff
@@ -427,12 +426,29 @@ export function mount (componentOptions: any, options: any = {}) {
         logInstance.end()
       }
 
-      // by returning undefined we keep the previous subject
-      // which is the mounted component
-      return {
-        wrapper,
-        component: wrapper.vm,
-      }
+      return new Proxy(Object.create({}), {
+        get (obj, prop) {
+          if (Reflect.get(wrapper, prop)) {
+            // @ts-expect-error - internal API
+            Cypress.utils.throwErrByPath('mount.vue_yielded_value')
+          }
+
+          if (prop === 'then') {
+            return {
+              wrapper,
+              component: wrapper.vm,
+            }
+          }
```

Comment:
Is the 'then' ever called?  The `chainable` will wrap this returned Proxy, but it will not have a then on it.